convention.md

```diff
@@ -18,11 +18,49 @@ The underscore (`_`) is used and reserved for Homie *node arrays*.
 
 ### Payload
 
-Every MQTT message payload MUST be sent as UTF-8 encoded string.
-A numeric data type MUST be converted to a literal string with the dot character as the decimal separator for the `float` data type.
-An empty message is not valid as payload for numeric data types.
-The value published as payload MUST be valid for the respective property/attribute type as per the convention definition below.
-Booleans MUST be converted to "true" or "false".
+ - Every MQTT message payload MUST be sent as a UTF-8 encoded string
+-  The value published as payload MUST be valid for the respective property/attribute type as per the defined convention
+ 
+String
+ - String types are limited to 268,435,456 characters 
+ - An empty string ("") is a valid payload
+ 
+Integer
+ - Integer types are UTF-8 encoded representations of 64-bit signed whole numbers
+ - Integers range from -9,223,372,036,854,775,808 (-2<sup>63</sup>) to 9,223,372,036,854,775,807 (2<sup>63</sup>-1)
+ - The payload may only contain whole numbers and the negation character "-". No other characters including spaces (" ") are permitted 
+ - A string with just a negation sign ("-") is not a valid payload
+ - An empty string ("") is not a valid payload
+ 
+Float
+ - Float types are UTF-8 encoded representations of 64-bit signed floating point numbers
+ - Floats range from 2<sup>-1074</sup> to (2-2<sup>-52</sup>)*2<sup>1023</sup>
+ - The payload may only contain whole numbers, the negation character "-", and the decimal separator ".", no other characters, including spaces (" ") are permitted 
+ - The dot character (".") is the decimal separator (used if necessary) and may only have a single instance present in the payload
+ - Representations of numeric concepts such as "NaN" (Not a Number) and "Infinity" are not a valid payload
+ - Representations using exponent notation ("e" or "E") are not valid payloads
+ - A string with just a negation sign ("-") is not a valid payload
+ - An empty string ("") is not a valid payload
+ 
+Boolean
+ - Booleans must be converted to "true" or "false"
+ - Representation is case sensitive, e.g. "TRUE" or "FALSE" are not valid payloads.
+ - An empty string ("") is not a valid payload
+ 
+Enum
+ - Enum payloads must be one of the values specified in the format definition of the property
+ - Enum payloads are case sensitive, e.g. "Car" will not match a format definition of "car"
+ - Enum matches are exact, e.g. A payload of "car ", will not match a format definition of "car". " car", "car" and "car " Represent three valid and distinct enum values
```

Comment:
What is the consensus on this?  This was my understanding of the spec as it is written today.  Do we want to allow spaces in enums? Do we want to support leading and trailing whitespace and have those be different enum values?

Reply:
Ok, I will update it to say that they aren't distinct enum value..
What languages with splitters by default strip whitespace?  Java doesn't, kotlin doesn't, ruby doesn't...

Reply:
In my opinion we shouldn't allow trailing or leading whitespace for enums.  I believe both the device and controller should trim all enum values.  It seems like supporting it would lead to errors with confusing troubleshooting with little to no gain.

Reply:
> C and C++ do :)

As far as I know, neither C nor C++ trims whitespaces if you split a string. But maybe I'm missing some API that does.
That aside, I think that whitespaces should not make a difference for matching. I.e. "car" = " car" = "car ".

Reply:
I will update the enum section to capture the feedback in this discussion.